components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -0,0 +1,914 @@
+
+    if (connected()) {
+      auto result_type = by_user
+          ? ads::NotificationResultInfoResultType::DISMISSED
+          : ads::NotificationResultInfoResultType::TIMEOUT;
+      bat_ads_->GenerateAdReportingNotificationResultEvent(
+          notification_info->ToJson(),
+          ToMojomNotificationResultInfoResultType(result_type),
+          base::BindOnce(&Noop));
+    }
+  }
+
+  std::move(completed_closure).Run();
+}
+
+void AdsServiceImpl::OpenSettings(Profile* profile,
+                                  const GURL& origin) {
+  DCHECK(origin.has_query());
+  auto notification_id = origin.query();
+
+  if (notification_ids_.find(notification_id) == notification_ids_.end())
+    return;
+
+  auto notification_info = base::WrapUnique(
+      notification_ids_[notification_id].release());
+  notification_ids_.erase(notification_id);
+
+  if (connected()) {
+    bat_ads_->GenerateAdReportingNotificationResultEvent(
+        notification_info->ToJson(),
+        ToMojomNotificationResultInfoResultType(
+            ads::NotificationResultInfoResultType::CLICKED),
+        base::BindOnce(&Noop));
+  }
+
+  GURL url(notification_info->url);
+
+  Browser* browser = chrome::FindLastActiveWithProfile(profile);
+  NavigateParams nav_params(browser, url, ui::PAGE_TRANSITION_LINK);
+  nav_params.disposition = WindowOpenDisposition::NEW_FOREGROUND_TAB;
+  // TODO(bridiver) - what to put here?
+  // nav_params.referrer = GURL("https://brave.com");
+  nav_params.window_action = NavigateParams::SHOW_WINDOW;
+  Navigate(&nav_params);
+}
+
+void AdsServiceImpl::GetClientInfo(ads::ClientInfo* client_info) const {
+  // TODO(bridiver) - these eventually get used in a catalog request
+  // and seem like potential privacy issues
+  // client_info->application_version = "";
+  // client_info->platform_version = "";
+  // client_info.application_version = chrome::kChromeVersion;
+#if defined(OS_MACOSX)
+  client_info->platform = ads::ClientInfoPlatformType::MACOS;
+#elif defined(OS_WIN)
+  client_info->platform = ads::ClientInfoPlatformType::WIN10;
+#elif defined(OS_LINUX)
+  client_info->platform = ads::ClientInfoPlatformType::LINUX;
+#elif defined(OS_ANDROID)
+  client_info->platform = ads::ClientInfoPlatformType::ANDROID;
+#else
+  NOTREACHED();
+  client_info->platform = ads::ClientInfoPlatformType::UNKNOWN;
+#endif
+}
+
+const std::string AdsServiceImpl::GenerateUUID() const {
+  return base::GenerateGUID();
+}
+
+const std::string AdsServiceImpl::GetSSID() const {
+    std::string ssid;
+#if defined(OS_WIN) || defined(OS_MACOSX)
+  std::unique_ptr<wifi::WiFiService> wifi_service(wifi::WiFiService::Create());
+  wifi_service->Initialize(nullptr);
+  std::string error;
+  wifi_service->GetConnectedNetworkSSID(&ssid, &error);
+  if (!error.empty())
+    return std::string();
+#elif defined(OS_LINUX)
+  ssid = net::GetWifiSSID();
+#elif defined(OS_ANDROID)
+  ssid = net::android::GetWifiSSID();
+#endif
+  // TODO: Handle non UTF8 SSIDs.
```

Comment:
Is this needed for 1.0?

Reply:
no, the comment was just copied from some chromium code as a reminder for when we deal with non-UTF8 everywhere

Reply:
it's also not critical information, mangled SSIDs won't impact anything important